cypress/e2e/initial-page-load.cy.ts

```diff
@@ -1,12 +1,38 @@
 context('Business Dashboard -> Basic page rendering tests', () => {
-  beforeEach(() => {
+  it('Loads the page with expected text', () => {
     cy.visitBusinessDash()
-  })
 
-  it('Loads the page with expected text', () => {
     cy.get('[data-cy="business-dashboard"]').should('exist')
 
     cy.get('[data-cy="business-dashboard"]').should('contain.text', 'To Do (0)')
     cy.get('[data-cy="business-dashboard"]').should('contain.text', 'Recent Filing History (0)')
   })
+
+  it('Loads indefinite, display loader icon', () => {
+    const legalType = 'BEN'
+    const isHistorical = false
+    const businessIdentifier = 'BC0871427'
+
+    sessionStorage.setItem('FAKE_CYPRESS_LOGIN', 'true')
+    cy.intercept('GET', '**/api/v1/users/**/settings', { fixture: 'settings.json' }).as('getSettings')
+    cy.intercept(
+      'REPORT',
+      'https://app.launchdarkly.com/sdk/evalx/**/context',
+      { fixture: 'ldarklyContext.json' }
+    ).as('getLdarklyContext')
+    cy.intercept('GET', '**/api/v1/orgs/**/products*', { fixture: 'products.json' }).as('getProducts')
+    cy.interceptBusinessContact(businessIdentifier, legalType).as('getBusinessContact')
+    cy.interceptBusinessInfo(businessIdentifier, legalType, isHistorical).as('getBusinessInfo')
+    cy.interceptAddresses(legalType).as('getAddresses')
+    cy.interceptParties(legalType, isHistorical).as('getParties')
+    cy.interceptAffiliationRequests(false, false).as('getAffiliationRequests')
+    cy.interceptTasks('tasksEmpty.json').as('getTasks')
+
+    cy.visit(`/${businessIdentifier}`)
+
+    cy.get('[data-cy="loading-icon"]').should('be.visible').then(() => {
+      cy.wait(['@getBusinessInfo'])
+      cy.get('[data-cy="loading-icon"]').should('not.exist')
```

Comment:
question: why do we just wait `@getBusinessInfo`? Do we need to wait for the other calls to be intercepted (like getTasks) before the loading icon disappears?

Reply:
No, mandatory one is to wait for 2nd load business call. Generally we could wait for others too, but the thing is there is ~5 sec wait anyways if it is not immediately available. And if it fails with timeout we should investigate why. So its not necessary here.